modules/gdscript/editor/script_templates/CompositorEffect/post_process_shader.gd

```diff
@@ -0,0 +1,149 @@
+		mutex.unlock()
+
+var rd: RenderingDevice
+var shader: RID
+var pipeline: RID
+
+var mutex: Mutex = Mutex.new()
+var shader_is_dirty: bool = true
+
+# Called when this resource is constructed.
+func _init():
+	effect_callback_type = EFFECT_CALLBACK_TYPE_POST_TRANSPARENT
+	rd = RenderingServer.get_rendering_device()
+
+# System notifications, we want to react on the notification that
+# alerts us we are about to be destroyed.
+func _notification(what):
+	if what == NOTIFICATION_PREDELETE:
+		if shader.is_valid():
+			# Freeing our shader will also free any dependents such as the pipeline!
+			RenderingServer.free_rid(shader)
+
+# Check if our shader has changed and needs to be recompiled.
+func _check_shader() -> bool:
+	if not rd:
+		return false
+
+	var new_shader_code: String = ""
+
+	# Check if our shader is dirty.
+	mutex.lock()
+	if shader_is_dirty:
+		new_shader_code = shader_code
+		shader_is_dirty = false
+	mutex.unlock()
+
+	# We don't have a (new) shader?
+	if new_shader_code.is_empty():
+		return pipeline.is_valid()
+
+	# Apply template.
+	new_shader_code = template_shader.replace("#COMPUTE_CODE", new_shader_code);
+
+	# Out with the old.
+	if shader.is_valid():
+		rd.free_rid(shader)
+		shader = RID()
+		pipeline = RID()
+
+	# In with the new.
+	var shader_source: RDShaderSource = RDShaderSource.new()
+	shader_source.language = RenderingDevice.SHADER_LANGUAGE_GLSL
+	shader_source.source_compute = new_shader_code
+	var shader_spirv: RDShaderSPIRV = rd.shader_compile_spirv_from_source(shader_source)
+
+	if shader_spirv.compile_error_compute != "":
+		push_error(shader_spirv.compile_error_compute)
+		push_error("In: " + new_shader_code)
+		return false
+
+	shader = rd.shader_create_from_spirv(shader_spirv)
+	if not shader.is_valid():
+		return false
+
+	pipeline = rd.compute_pipeline_create(shader)
+	return pipeline.is_valid()
+
+# Called by the rendering thread every frame.
+func _render_callback(p_effect_callback_type, p_render_data):
+	if rd and p_effect_callback_type == EFFECT_CALLBACK_TYPE_POST_TRANSPARENT and _check_shader():
+		# Get our render scene buffers object, this gives us access to our render buffers.
+		# Note that implementation differs per renderer hence the need for the cast.
+		var render_scene_buffers: RenderSceneBuffersRD = p_render_data.get_render_scene_buffers()
+		if render_scene_buffers:
+			# Get our render size, this is the 3D render resolution!
+			var size = render_scene_buffers.get_internal_size()
+			if size.x == 0 and size.y == 0:
+				return
+
+			# We can use a compute shader here
+			var x_groups = (size.x - 1) / 8 + 1
+			var y_groups = (size.y - 1) / 8 + 1
+			var z_groups = 1
+
+			# Push constant
+			var push_constant: PackedFloat32Array = PackedFloat32Array()
+			push_constant.push_back(size.x)
+			push_constant.push_back(size.y)
+			push_constant.push_back(0.0)
+			push_constant.push_back(0.0)
+
+			# Loop through views just in case we're doing stereo rendering. No extra cost if this is mono.
```

Comment:
This line exceeds the [recommended](https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_styleguide.html#line-length) 100 character limit.